SEED-platform/seed
# 2345 fix/manual edit saving

#### Any background context you want to provide?
When editing inventory manually on details page, you could change nothing and still click save. Posting empty changes would make the server 500.
#### What's this PR do?
- Fixes 500 issue by returning correct number of values in utility func. Specifically, in `seed/utils/properties.py` function `get_changed_fields` was returning a single `None` if nothing was changed, while the caller (the update view) expected two returned values resulting in `TypeError: 'NoneType' object is not iterable`
- Fixes frontend issue of detecting changes when none really exist. Specifically, in `seed/static/seed/js/controllers/inventory_detail_controller.js`, the function `$scope.diff` was returning non-empty diff's when nothing was changed (it had props `files`, `measures`, and `scenarios` containing empty arrays). `$scope.modified` checks if the return value of `diff` is empty, which is used to disable the "Save" button when nothing has been edited.
#### How should this be manually tested?
- Upload taxlot/properties. click edit, you should not be able to click save until you change something. Click save and it should save properly
#### What are the relevant tickets?
#2345
